kivy/core/text/__init__.py

```diff
@@ -604,7 +605,16 @@ def render(self, real=False):
         if strip:
             text = text.strip()
         if uw is not None and options['shorten']:
-            text = self.shorten(text)
+            short_text = self.shorten(text)
+            chr = type(text)
+            flat_text = text.replace(chr('\n'), chr(' '))
+            if flat_text != short_text:
```

Comment:
I think the reason behind this was to compare flattened and shortened text, but the already mentioned multiple returns would make it longer to write (for each return?) in the markup itself and maybe not even faster. Not sure how to handle this.